test/integration/default/controls/_mapdata_spec.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# Keep only first 2 digits from Ubuntu finger
+mapdata_file = "_mapdata/#{system.platform[:finger].split('.').first}.yaml"
+
+# Load the mapdata from profile https://docs.chef.io/inspec/profiles/#profile-files
+mapdata_dump = inspec.profile.file(mapdata_file)
+
+control '`map.jinja` YAML dump' do
+  title 'should contain the lines'
+
+  describe file('/tmp/salt_mapdata_dump.yaml').content.encode(universal_newline: true) do # rubocop:disable Layout/LineLength
+    it { should eq mapdata_dump }
+  end
```

Comment:
@baby-gnu @dafyddj New proposed method, to stay close to the existing structure.

Reply:
Just a comment on the output of the check using this method - it's rather verbose (it always prints out the whole YAML tree) compared with typical inspec checks. 
It's possible to change that using `expect` as shown [here](https://github.com/saltstack-formulas/openssh-formula/pull/193#issuecomment-709445531), inspired by [this](https://github.com/inspec/inspec/issues/2310#issuecomment-344742155).

Reply:
@dafyddj Hmm, that `expect` method isn't working out too well at my end -- it's not showing the actual problem (diff):

```yaml
  ×  `map.jinja` YAML dump: should contain the lines
     ×  File content should match profile map data exactly
     
     expected: "# yamllint disable rule:indentation rule:line-length\n# Windows-8.1\n---\nbin_dir: C:\\Program Files...penvpn-gnome\npkgs:\n- openvpns\nservice: OpenVPNService\nservice_function: running\nuser: nobody\n"
          got: "# yamllint disable rule:indentation rule:line-length\n# Windows-8.1\n---\nbin_dir: C:\\Program Files...openvpn-gnome\npkgs:\n- openvpn\nservice: OpenVPNService\nservice_function: running\nuser: nobody\n"
     
     (compared using ==)
```

Reply:
@dafyddj Good news!  Limiting the gem has also caused the diff to work with `expect` and that also achieves the goal of suppressing all of the excess output when tests pass.